src/garage/torch/algos/ppo.py

```diff
@@ -0,0 +1,96 @@
+"""Proximal Policy Optimization (PPO)."""
+import torch
+
+from garage.torch.algos import VPG
+
+
+class PPO(VPG):
+    """Proximal Policy Optimization (PPO).
+
+    Args:
+        env_spec (garage.envs.EnvSpec): Environment specification.
+        policy (garage.torch.policies.base.Policy): Policy.
+        baseline : The baseline.
+        max_path_length (int): Maximum length of a single rollout.
+        policy_lr (float): Learning rate for training policy network.
+        lr_clip_range (float): The limit on the likelihood ratio between
+            policies.
+        n_samples (int): Number of train_once calls per epoch.
+        discount (float): Discount.
+        gae_lambda (float): Lambda used for generalized advantage
+            estimation.
+        center_adv (bool): Whether to rescale the advantages
+            so that they have mean 0 and standard deviation 1.
+        positive_adv (bool): Whether to shift the advantages
+            so that they are always positive. When used in
+            conjunction with center_adv the advantages will be
+            standardized before shifting.
+        optimizer (object): The optimizer of the algorithm. Should be the
+            optimizers in torch.optim.
+        policy_ent_coeff (float): The coefficient of the policy entropy.
+            Setting it to zero would mean no entropy regularization.
+        use_softplus_entropy (bool): Whether to estimate the softmax
+            distribution of the entropy to prevent the entropy from being
+            negative.
+        stop_entropy_gradient (bool): Whether to stop the entropy gradient.
+        entropy_method (str): A string from: 'max', 'regularized',
+            'no_entropy'. The type of entropy method to use. 'max' adds the
+            dense entropy to the reward for each time step. 'regularized' adds
+            the mean entropy to the surrogate objective. See
+            https://arxiv.org/abs/1805.00909 for more details.
+    """
+
+    def __init__(self,
+                 env_spec,
+                 policy,
+                 baseline,
+                 max_path_length=500,
+                 policy_lr=1e-2,
+                 lr_clip_range=1e-2,
+                 n_samples=1,
+                 discount=0.99,
+                 gae_lambda=1,
+                 center_adv=True,
+                 positive_adv=False,
+                 optimizer=None,
+                 policy_ent_coeff=0.0,
+                 use_softplus_entropy=False,
```

Comment:
several of these seem unused?

Reply:
@yonghyuc I agree with ryan, it seems a lot of these are not used

Reply:
Currently some of them (like `use_softplus_entropy` and `entropy_method`) are used in the parent class (VPG).
Is there no case that people use the parameters in PPO? 

@ryanjulian @avnishn

Reply:
these are used to add entropy regularization, which you haven't implemented here. It would be nice to have entropy, but you can implement it next (or add it to this PR).

Did you implement entropy in VPG already? If you did, you should implement it in PPO as well.

Don't include constructor parameters which aren't used, even if the parent constructor has them.

Reply:
I passed the `use_softplus_entropy` and `entropy_method` to (parent) VPG class. 
Since PPO only override `_compute_objective` function which is called inside `_compute_loss` (in VPG), these paremeters would used during `_compute_loss`.
You can check [here](https://github.com/rlworkgroup/garage/blob/363ebeb326916c89a1ffa25b1f63eeafdbf0eaf7/src/garage/torch/algos/vpg.py#L135) and [here](https://github.com/rlworkgroup/garage/blob/363ebeb326916c89a1ffa25b1f63eeafdbf0eaf7/src/garage/torch/algos/vpg.py#L158)